tls/s2n_security_policies.c

```diff
@@ -51,6 +51,15 @@ const struct s2n_security_policy security_policy_default_fips = {
     .ecc_preferences = &s2n_ecc_preferences_default_fips,
 };
 
+const struct s2n_security_policy security_policy_20230317 = {
+    .minimum_protocol_version = S2N_TLS12,
+    .cipher_preferences = &cipher_preferences_20230317,
+    .kem_preferences = &kem_preferences_null,
+    .signature_preferences = &s2n_signature_preferences_20230317,
+    .certificate_signature_preferences = &s2n_signature_preferences_20230317,
+    .ecc_preferences = &s2n_ecc_preferences_default_fips,
```

Comment:
this has p256 and p384, which is probably fine. Technically p521 is allowed too. https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-52r2.pdf refers to https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-56Ar3.pdf Appendix D which lists secp521r1.

Reply:
I forgot p521 even exists :) Are you suggesting including p521 for this policy?

Reply:
I don't see the harm. It's FIPS approved curve. it's not slower than p384 based on testing done. If a client only supports p521 it will avoid handshake failures.

Reply:
(other argument is that we allow p521 for signatures)